### DISCUSSION TITLE
역의 변동 가능한 속성 값 관련

### DISCUSSION THREAD

**Discussion by mille-feu** [Tiny-Metro/Poly-Metro#208] [Q&A]
이벤트, 정책, 미션 등으로 변경 가능한 값들의 예시는 다음과 같습니다.
- 승객 생성 속도
- 승객 생성 확률
- 불만도 증감 가중치
- 무료 승객 생성 확률
- 기타

위와 같은 값들은, 변동 이후 생성되는 역에도 적용이 되나요?
아니면 특정 경우에 따라 각각 다르게 적용되나요?

**Comment by cstrnull00** [comment]
현재 계획 중인 이벤트/정책/미션 등지에서 생성 시점에 따라 역에 다르게 적용되는 설정은 없습니다.
단, 역의 모양이나 위치, 노선, 등에 따라 해당 수치가 독립적으로 적용될 가능성은 있습니다.

ex) 
- 특정 이벤트/정책/미션을 통해 승객 생성 속도, 확률 등의 가중치가 맵 전체에 적용되었을 경우, 새 역이 생성되었을 때 : 동일한 가중치 적용
- 특정 이벤트/정책/미션이 역의 모양, 위치, 노선에 한정되어 적용될 경우, 새 역이 생성되었을 때 : 새 역이 해당 요건을 만족할 경우 동일한 가중치 적용

**Comment by mille-feu** [comment]
추가적인 질문입니다.
위 값들에 대하여, 현재 API에서 제공하는 기능은 다음과 같습니다.
- 일정 기간 수치의 변화 (곱연산) (Add)
- 일정 기간 수치의 변화 (합연산) (Multiple)
- 값 자체의 수정 (Set)

현재 적용된 곱, 합연산의 구현은 다음과 같습니다.
- 연산 수치를 배열로 저장 (곱, 합 별도)
- 배열의 수치를 각각 곱(합)하여 최종 적용 값 계산
- 기존 수치 x 곱연산 최종 수치 + 합연산 최종 수치 의 연산을 통해 최종 값 적용

논의가 필요한 내용은 Set연산의 필요성 및 적용 방식입니다.
- Set연산이 필요할 경우
  - Set연산 적용 시, 기존의 곱, 합 연산 수치의 처리 여부
  - 적용 기간의 유무
- Set연산이 필요하지 않을 경우
  - API삭제

**Reply by cstrnull00** [reply]
이벤트/미션 등지에서는 set 연산을 사용하지 않을 것 같습니다. 삭제하셔도 무방합니다.
